tools/cli/NearObjectCli.cxx

```diff
@@ -292,3 +317,30 @@ NearObjectCli::AddSubcommandUwbRangeStop(CLI::App* parent)
 
     return rangeStopApp;
 }
+
+CLI::App*
+NearObjectCli::AddSubcommandUwbRawGetDeviceInfo(CLI::App* parent)
+{
+    // top-level command
+    auto rawGetDeviceInfoApp = parent->add_subcommand("getdeviceinfo", "GetDeviceInfo")->fallthrough();
+
+    rawGetDeviceInfoApp->parse_complete_callback([this] {
+        std::cout << "get device info" << std::endl;
+    });
+
+    rawGetDeviceInfoApp->final_callback([this] {
+        auto uwbDevice = GetUwbDevice();
+        if (!uwbDevice) {
+            std::cerr << "no device found" << std::endl;
+            return;
+        }
+        
+        if (uwbDevice->Initialize()) {
+            m_cliHandler->HandleGetDeviceInfo(uwbDevice);
+        } else {
+            std::cerr << "device not initialized" << std::endl;
+        }
```

Comment:
To keep with the prevailing pattern, I think we should return early when an error is recognized to redeuce unnecessary nested scoping.

```suggestion
        if (!uwbDevice->Initialize()) {
            std::cerr << "device not initialized" << std::endl;
            return;
        }
        
        m_cliHandler->HandleGetDeviceInfo(uwbDevice);
```